config/smartyfront.config.inc.php

```diff
@@ -59,15 +59,24 @@
 
 function withWidget($params, callable $cb)
 {
-    if (!isset($params['name'])) {
-        throw new Exception('Smarty helper `render_widget` expects at least the `name` parameter.');
+    // Check if name was provided
+    if (empty($params['name'])) {
+        throw new Exception('When using {widget}, you must provide at least the `name` parameter.');
     }
 
+    // Get module name
     $moduleName = $params['name'];
     unset($params['name']);
 
+    // Try to load module
     $moduleInstance = Module::getInstanceByName($moduleName);
 
+    // If it's not installed, nothing to do here
+    if (empty($moduleInstance)) {
+        return;
```

Comment:
Suggesting an alternative instead of returning completely silenty, thus totally hiding that something is wrong without the develop knowing, we should throw an exception when debug mode is on and log a warning or error when in production mode

The legacy code is full of such cases where we simply ignore the errors, which makes it really hard to understand the root cause of any bug

Reply:
@jolelievre maybe an Exception if in the debug mode?